Patient Registry System/PatientRegistrySystem.API/Controllers/AdminController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using PatientRegistrySystem.API.ViewModel;
+using PatientRegistrySystem.API.ViewModel.Registration;
+using PatientRegistrySystem.DB.Entities;
+using PatientRegistrySystem.Domain.Dto;
+using PatientRegistrySystem.Services;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientRegistrySystem.API.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    [Authorize(Roles = "Admin")]
+    public class AdminController : Controller
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IMapper _mapper;
+        private readonly IUserService _userService;
+
+        public AdminController(UserManager<ApplicationUser> userManager,
+             IMapper mapper,
+            IUserService userService)
+        {
+            _userManager = userManager;
+            _mapper = mapper;
+            _userService = userService;
+        }
+
+        [Route("[action]")]
+        [HttpPost]
+        [Authorize(Roles = "Admin")]
```

Comment:
for the roles, lets add a static file with the role names:
public class static UserRoles
{
public static string Admin = "Admin";
....
}